soh/src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -1675,7 +1688,18 @@ void EnOssan_State_GiveItemWithFanfare(EnOssan* this, GlobalContext* globalCtx,
         this->stateFlag = OSSAN_STATE_ITEM_PURCHASED;
         return;
     }
-    func_8002F434(&this->actor, globalCtx, this->shelfSlots[this->cursorIndex]->getItemId, 120.0f, 120.0f);
+    if (!gSaveContext.n64ddFlag) {
```

Comment:
Nice catch, missed this one

Reply:
Yep this is what caused my confusion last night with the item fanfare having the wrong item lol. There was one earlier that got called when buying an item the first time and this one got called when buying the item after the "Would you like to continue shopping" prompt lol.